src/Tracing/Transaction.php

```diff
@@ -59,6 +65,29 @@ public function setName(string $name): void
         $this->name = $name;
     }
 
+    /**
+     * Gets the transaction metadata.
+     */
+    public function getMetadata(): TransactionMetadata
+    {
+        return $this->metadata;
+    }
+
+    /**
+     * Gets the transaction dynamic sampling context.
+     */
+    public function getDynamicSamplingContext(): DynamicSamplingContext
+    {
+        if (null !== $this->getMetadata()->getDynamicSamplingContext()) {
+            return $this->getMetadata()->getDynamicSamplingContext();
+        }
+
+        $dsc = DynamicSamplingContext::fromTransaction($this->transaction, $this->hub);
```

Comment:
Given the `$this->metadata = $context->getMetadata();` call in the constructor, do we might need to add some annotation to please phpstan?

Reply:
As the `Span::class` contains a property `Transaction|null`, I added it to the Transaction as well, annotating it as `Transaction`. Not sure if this is a good approach, though.

Reply:
This is what I would propose to do here, it cannot be null on the transaction class so makes sense to override with a new annotation.

Reply:
Great. I already added the new property and will add a template annotation to please psalm as well.